src/Concordium/Client/Runner.hs

```diff
@@ -537,15 +561,22 @@ processTransactionCmd action baseCfgDir verbose backend =
       -- TODO This works but output doesn't make sense if transaction is already committed/finalized.
       --      It should skip already completed steps.
       -- withClient backend $ tailTransaction_ hash
-    TransactionSendGtu receiver amount txOpts -> do
+    TransactionSendGtu receiver amount maybeMemo txOpts -> do
       baseCfg <- getBaseConfig baseCfgDir verbose
       when verbose $ do
         runPrinter $ printBaseConfig baseCfg
         putStrLn ""
 
       receiverAddress <- getAccountAddressArg (bcAccountNameMap baseCfg) $ Just receiver
 
-      let pl = Types.encodePayload $ Types.Transfer (naAddr receiverAddress) amount
+      let plDo = do
+            res <- case maybeMemo of 
+              Nothing -> return $ Types.Transfer (naAddr receiverAddress) amount
+              Just memo -> do
+                bs <- readMemoAsBytestring memo backend
+                return $ Types.TransferWithMemo (naAddr receiverAddress) (Types.Memo $ BSS.toShort bs) amount 
+            return $ Types.encodePayload res
+      pl <- plDo
```

Comment:
You do not have to introduce `plDo` (if you do not want to), e.g., you can write

```haskell
pl <- do
   res <- case maybeMemo of
      ...
   return $ Types.encodePayload res
```

If you change the `readMemoAsBytes` to live in the ClientMonad then you will need to restructure this a little bit by doing

```haskell
withClient backend $ do
   let plDo = do
       ...
   pl <- plDo
   let nrgCost = ...
   ...
   sendAndTailTransaction_ txCfg pl intOpts
```